app/controllers/houseConsignment/index/items/additionalReference/AdditionalReferenceTypeController.scala

```diff
@@ -104,31 +102,26 @@ class AdditionalReferenceTypeController @Inject() (
                       view(formWithErrors, request.userAnswers.mrn, additionalReferences.values, viewModel)
                     )
                   ),
-                value => redirect(value, houseConsignmentIndex, itemIndex, additionalReferenceIndex, houseConsignmentMode, itemMode, additionalReferenceMode)
+                value =>
+                  for {
```

Comment:
Just wondering if we need to be storing this value, I think we might be able to keep this controller as it was, but inside of the AdditionalReferenceNumberController simply check to see if the value is in the list?

e.g
```
package controllers.houseConsignment.index.items.additionalReference

import controllers.actions._
import forms.ItemsAdditionalReferenceNumberFormProvider
import generated.AdditionalReferenceType02
import models.reference.AdditionalReferenceType
import models.requests.MandatoryDataRequest
import models.{ArrivalId, Index, Mode}
import navigation.houseConsignment.index.items.AdditionalReferenceNavigator.AdditionalReferenceNavigatorProvider
import pages.houseConsignment.index.items.additionalReference.{AdditionalReferenceInCL234Page, AdditionalReferenceNumberPage, AdditionalReferenceTypePage}
import play.api.i18n.{I18nSupport, MessagesApi}
import play.api.mvc.{Action, AnyContent, MessagesControllerComponents, Result}
import repositories.SessionRepository
import services.AdditionalReferencesService
import uk.gov.hmrc.http.HeaderCarrier
import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
import viewModels.houseConsignment.index.items.additionalReference.AdditionalReferenceNumberViewModel.AdditionalReferenceNumberViewModelProvider
import views.html.houseConsignment.index.items.additionalReference.AdditionalReferenceNumberView

import javax.inject.Inject
import scala.concurrent.{ExecutionContext, Future}

class AdditionalReferenceNumberController @Inject() (
  override val messagesApi: MessagesApi,
  sessionRepository: SessionRepository,
  navigatorProvider: AdditionalReferenceNavigatorProvider,
  formProvider: ItemsAdditionalReferenceNumberFormProvider,
  actions: Actions,
  getMandatoryPage: SpecificDataRequiredActionProvider,
  val controllerComponents: MessagesControllerComponents,
  view: AdditionalReferenceNumberView,
  viewModelProvider: AdditionalReferenceNumberViewModelProvider,
  service: AdditionalReferencesService,
                                                    )(implicit ec: ExecutionContext)
    extends FrontendBaseController
    with I18nSupport {

  def isInCL234(documentType: AdditionalReferenceType)(implicit hc: HeaderCarrier): Future[Boolean] =
    service.isDocumentTypeExcise(documentType.documentType)

  def onPageLoad(
    arrivalId: ArrivalId,
    houseConsignmentMode: Mode,
    itemMode: Mode,
    additionalReferenceMode: Mode,
    houseConsignmentIndex: Index,
    itemIndex: Index,
    additionalReferenceIndex: Index
  ): Action[AnyContent] =
    actions
      .requireData(arrivalId)
      .andThen(getMandatoryPage(AdditionalReferenceTypePage(houseConsignmentIndex, itemIndex, additionalReferenceIndex)))
      .async
      {
        implicit request =>
          val viewModel = viewModelProvider.apply(
            arrivalId,
            houseConsignmentMode,
            itemMode,
            additionalReferenceMode,
            houseConsignmentIndex,
            itemIndex,
            additionalReferenceIndex
          )

          isInCL234(request.arg).map {
            isInCL234 =>
            val form = formProvider(viewModel.requiredError, isInCL234)
            val preparedForm = request.userAnswers.get(AdditionalReferenceNumberPage(houseConsignmentIndex, itemIndex, additionalReferenceIndex)) match {
              case None        => form
              case Some(value) => form.fill(value)
            }
            Ok(view(preparedForm, request.userAnswers.mrn, viewModel))
          }

      }

  def onSubmit(
    arrivalId: ArrivalId,
    houseConsignmentMode: Mode,
    itemMode: Mode,
    additionalReferenceMode: Mode,
    houseConsignmentIndex: Index,
    itemIndex: Index,
    additionalReferenceIndex: Index
  ): Action[AnyContent] = actions
    .requireData(arrivalId)
    .andThen(getMandatoryPage(AdditionalReferenceTypePage(houseConsignmentIndex, itemIndex, additionalReferenceIndex)))
    .async {
      implicit request =>
        val viewModel = viewModelProvider.apply(
          arrivalId,
          houseConsignmentMode,
          itemMode,
          additionalReferenceMode,
          houseConsignmentIndex,
          itemIndex,
          additionalReferenceIndex
        )
        isInCL234(request.arg).flatMap {
          isInCL234 =>
            val form = formProvider(viewModel.requiredError, isInCL234)
            form
              .bindFromRequest()
              .fold(
                formWithErrors => Future.successful(BadRequest(view(formWithErrors, request.userAnswers.mrn, viewModel))),
                value => redirect(value, houseConsignmentIndex, itemIndex, additionalReferenceIndex, houseConsignmentMode, itemMode, additionalReferenceMode)
              )
        }
    }

  private def redirect(
    value: String,
    houseConsignmentIndex: Index,
    itemIndex: Index,
    additionalReferenceIndex: Index,
    houseConsignmentMode: Mode,
    itemMode: Mode,
    additionalReferenceMode: Mode
  )(implicit request: MandatoryDataRequest[_]): Future[Result] =
    for {
      updatedAnswers <- Future.fromTry(
        request.userAnswers.set(AdditionalReferenceNumberPage(houseConsignmentIndex, itemIndex, additionalReferenceIndex), value)
      )
      _ <- sessionRepository.set(updatedAnswers)
    } yield {
      val navigator = navigatorProvider.apply(houseConsignmentMode, itemMode)
      Redirect(
        navigator.nextPage(AdditionalReferenceNumberPage(houseConsignmentIndex, itemIndex, additionalReferenceIndex),
                           additionalReferenceMode,
                           request.userAnswers
        )
      )
    }
}
```